Assets/Magic/Mana.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Mana : MonoBehaviour
+{
+    [SerializeField] private float maxMana;
+
+    private float _mana;
+    private void Start()
+    {
+        _mana = maxMana;
+    }
+
+    /// <summary>
+    /// Try to use given amount of mana.
+    /// </summary>
+    /// <param name="expendedMana">The amount of mana to subtract from the mana pool.</param>
+    /// <returns>True if succeeded, false otherwise.</returns>
+    /// <example>
+    /// <code>
+    /// bool succeeded = UseMana(50f);
+    /// if (!succeeded) return; // There wasn't enough mana.
+    /// // Use magic
+    /// </code>
+    /// </example>
+    public bool UseMana(float expendedMana)
+    {
+        if (_mana - expendedMana < 0f) return false;
```

Comment:
```suggestion
        if (_mana - expendedMana < minimaleMana) return false;
```
ipv een 0f neer te zetten, boven aan een [SerializeField] float neer te zetten.